gendo/bot.py

```diff
@@ -31,10 +31,12 @@ def config_from_yaml(cls, path_to_yaml):
             settings = yaml.load(ymlfile)
             return cls(settings=settings)
 
-    def listen_for(self, rule, **options):
+    def listen_for(self, rule):
         def decorator(f):
-            self.add_listener(rule, f, **options)
-            return f
+            def wrapped(**kwargs):
+                self.add_listener(rule, f, **kwargs)
+                return f
+            wrapped()
```

Comment:
Just pass any named argument to your original decorator maker 'listen_for' method. It will be added to **options hash, then will be passed to 'listeners' array via 'add_listener'.  After this you will do `view_func(user, message, **options)` in 'respond' method. But 'view_func' is _original_ function, not decorator. So, as a result - it will fail, cause in function you decorate you won't have a signature with **options.